### PR TITLE
[DT-427] Updated project libraries

### PR DESCRIPTION
**Pull request by ricardo-frango**
**What does this PR do?**

   This PR was created to upgrade the libraries versions.

**Database changed?**

   No

**How should this be manually tested?**

  Flow on how to test this or QA Tickets related to this use-case: [DT-427](https://aptoide.atlassian.net/browse/DT-427)

**What are the relevant tickets?**

  Tickets related to this pull-request: [DT-427](https://aptoide.atlassian.net/browse/DT-427)

**Code Review Checklist**

- [ ] Documentation on public interfaces
- [ ] Database changed? If yes - Migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] New Kotlin code has unit tests
- [ ] New flows in presenters unit tests
- [ ] Mappers/Validators with any kind of logic unit tests
- [ ] Functional tests pass

[DT-427]: https://aptoide.atlassian.net/browse/DT-427?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DT-427]: https://aptoide.atlassian.net/browse/DT-427?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: aptoide-ui/build.gradle.kts [hunk 1]
```diff
@@ -28,15 +28,14 @@ android {
   buildFeatures {
     // Enables Jetpack Compose for this module
     compose = true
+    buildConfig = true
```
Comment (G0retZ) [inline aptoide-ui/build.gradle.kts:31]: No need in this, default is `true`

File: download-view/build.gradle.kts [hunk 2]
```diff
@@ -28,15 +28,14 @@ android {
   buildFeatures {
     // Enables Jetpack Compose for this module
     compose = true
+    buildConfig = true
```
Comment (G0retZ) [inline download-view/build.gradle.kts:31]: No need in this, default is `true`

File: feature-home/build.gradle.kts [hunk 3]
```diff
@@ -30,14 +30,12 @@ android {
   buildFeatures {
     // Enables Jetpack Compose for this module
     compose = true
+    buildConfig = true
```
Comment (G0retZ) [inline feature-home/build.gradle.kts:33]: No need in this, default is `true`

File: feature-profile/build.gradle.kts [hunk 4]
```diff
@@ -30,14 +30,12 @@ android {
   buildFeatures {
     // Enables Jetpack Compose for this module
     compose = true
+    buildConfig = true
```
Comment (G0retZ) [inline feature-profile/build.gradle.kts:33]: No need in this, default is `true`

File: feature-reactions/build.gradle.kts [hunk 5]
```diff
@@ -30,14 +30,12 @@ android {
   buildFeatures {
     // Enables Jetpack Compose for this module
     compose = true
+    buildConfig = true
```
Comment (G0retZ) [inline feature-reactions/build.gradle.kts:33]: No need in this, default is `true`

File: feature_apps/build.gradle.kts [hunk 6]
```diff
@@ -30,14 +30,12 @@ android {
   buildFeatures {
     // Enables Jetpack Compose for this module
     compose = true
+    buildConfig = true
```
Comment (G0retZ) [inline feature_apps/build.gradle.kts:33]: No need in this, default is `true`

File: feature_apps/src/main/java/cm/aptoide/pt/feature_apps/presentation/AppGraphicView.kt
Comment (G0retZ) [inline feature_apps/src/main/java/cm/aptoide/pt/feature_apps/presentation/AppGraphicView.kt:1]: An unrelated changes in this file
Should be resolve in another ticket or in a dedicated formatting commit

File: feature_apps/src/main/java/cm/aptoide/pt/feature_apps/presentation/AppGridView.kt
Comment (G0retZ) [inline feature_apps/src/main/java/cm/aptoide/pt/feature_apps/presentation/AppGridView.kt:1]: An unrelated changes in this file
Should be resolve in another ticket or in a dedicated formatting commit

File: feature_editorial/build.gradle.kts [hunk 7]
```diff
@@ -30,14 +30,12 @@ android {
   buildFeatures {
     // Enables Jetpack Compose for this module
     compose = true
+    buildConfig = true
```
Comment (G0retZ) [inline feature_editorial/build.gradle.kts:33]: No need in this, default is `true`

File: buildSrc/src/main/kotlin/CoreVersion.kt [hunk 8]
```diff
@@ -1,6 +1,6 @@
 object CoreVersion {
-  const val KOTLIN = "1.6.21"
-  const val COROUTINES = "1.6.0"
-  const val NAVIGATION = "2.3.5"
-  const val KT_COMPILER_EXTENSION = "1.2.0-rc02"
+  const val KOTLIN = "1.8.21"
+  const val COROUTINES = "1.7.1"
+  const val NAVIGATION = "2.5.3"
+  const val KT_COMPILER_EXTENSION = "1.4.7"
 }
```
Comment (G0retZ) [inline buildSrc/src/main/kotlin/CoreVersion.kt:6]: New line is missing

File: feature_appview/src/main/java/cm/aptoide/pt/feature_appview/presentation/RelatedContentView.kt
Comment (G0retZ) [inline feature_appview/src/main/java/cm/aptoide/pt/feature_appview/presentation/RelatedContentView.kt:1]: An unrelated changes in this file
Should be resolve in another ticket or in a dedicated formatting commit

File: feature_editorial/src/main/java/cm/aptoide/pt/feature_editorial/presentation/EditorialsCardView.kt
Comment (G0retZ) [inline feature_editorial/src/main/java/cm/aptoide/pt/feature_editorial/presentation/EditorialsCardView.kt:1]: An unrelated changes in this file
Should be resolve in another ticket or in a dedicated formatting commit

File: feature_report_app/src/main/java/cm/aptoide/pt/feature_report_app/presentation/ReportAppView.kt
Comment (G0retZ) [inline feature_report_app/src/main/java/cm/aptoide/pt/feature_report_app/presentation/ReportAppView.kt:1]: An unrelated changes in this file
Should be resolve in another ticket or in a dedicated formatting commit

File: feature_search/src/main/java/cm/aptoide/pt/feature_search/data/database/SearchHistoryDao.kt
Comment (G0retZ) [inline feature_search/src/main/java/cm/aptoide/pt/feature_search/data/database/SearchHistoryDao.kt:1]: An unrelated changes in this file
Should be resolve in another ticket or in a dedicated formatting commit

File: install-manager/build.gradle.kts [hunk 9]
```diff
@@ -2,10 +2,10 @@ plugins {
   id(GradlePluginId.ANDROID_LIBRARY)
   id(GradlePluginId.KOTLIN_ANDROID)
   id(GradlePluginId.JUNIT5_PLUGIN)
-  id(GradlePluginId.KOTLIN_KAPT)
 }
 
 android {
+  namespace = "cm.aptoide.pt.app_installer"
```
Comment (G0retZ) [inline install-manager/build.gradle.kts:8]: Why this namespace? 🤔

File: install-manager/src/main/AndroidManifest.xml
Comment (G0retZ) [inline install-manager/src/main/AndroidManifest.xml:1]: An unrelated changes in this file
Should be resolve in another ticket or in a dedicated formatting commit

File: app-dt
Comment (G0retZ) [inline app-dt:1]: Do not commit this please

File: app/build.gradle.kts [hunk 10]
```diff
@@ -37,6 +37,7 @@ android {
   buildFeatures {
     // Enables Jetpack Compose for this module
     compose = true
+    buildConfig = true
```
[the remark this answers is not in the document]
Comment (ricardo-frango) [reply app/build.gradle.kts:40]: Apparently on the the new AGP 8.0 the default value has been changed to false. I had to change it to true.

https://developer.android.com/reference/tools/gradle-api/8.0/com/android/build/api/dsl/BuildFeatures#buildConfig()

File: aptoide-ui/build.gradle.kts (hunk 1 above)
Comment (ricardo-frango) [reply aptoide-ui/build.gradle.kts:31]: Apparently on the the new AGP 8.0 the default value has been changed to false. I had to change it to true.

https://developer.android.com/reference/tools/gradle-api/8.0/com/android/build/api/dsl/BuildFeatures#buildConfig()

File: feature-home/build.gradle.kts (hunk 3 above)
Comment (ricardo-frango) [reply feature-home/build.gradle.kts:33]: Apparently on the the new AGP 8.0 the default value has been changed to false. I had to change it to true.

https://developer.android.com/reference/tools/gradle-api/8.0/com/android/build/api/dsl/BuildFeatures#buildConfig()

File: feature-profile/build.gradle.kts (hunk 4 above)
Comment (ricardo-frango) [reply feature-profile/build.gradle.kts:33]: Apparently on the the new AGP 8.0 the default value has been changed to false. I had to change it to true.

https://developer.android.com/reference/tools/gradle-api/8.0/com/android/build/api/dsl/BuildFeatures#buildConfig()

File: feature-reactions/build.gradle.kts (hunk 5 above)
Comment (ricardo-frango) [reply feature-reactions/build.gradle.kts:33]: Apparently on the the new AGP 8.0 the default value has been changed to false. I had to change it to true.

https://developer.android.com/reference/tools/gradle-api/8.0/com/android/build/api/dsl/BuildFeatures#buildConfig()

File: feature_apps/build.gradle.kts (hunk 6 above)
Comment (ricardo-frango) [reply feature_apps/build.gradle.kts:33]: Apparently on the the new AGP 8.0 the default value has been changed to false. I had to change it to true.

https://developer.android.com/reference/tools/gradle-api/8.0/com/android/build/api/dsl/BuildFeatures#buildConfig()

File: feature_appview/build.gradle.kts [hunk 11]
```diff
@@ -28,15 +29,14 @@ android {
   buildFeatures {
     // Enables Jetpack Compose for this module
     compose = true
+    buildConfig = true
```
[the remark this answers is not in the document]
Comment (ricardo-frango) [reply feature_appview/build.gradle.kts:32]: Apparently on the the new AGP 8.0 the default value has been changed to false. I had to change it to true.

https://developer.android.com/reference/tools/gradle-api/8.0/com/android/build/api/dsl/BuildFeatures#buildConfig()

File: feature_report_app/build.gradle.kts [hunk 12]
```diff
@@ -31,14 +31,12 @@ android {
   buildFeatures {
     // Enables Jetpack Compose for this module
     compose = true
+    buildConfig = true
```
[the remark this answers is not in the document]
Comment (ricardo-frango) [reply feature_report_app/build.gradle.kts:34]: Apparently on the the new AGP 8.0 the default value has been changed to false. I had to change it to true.

https://developer.android.com/reference/tools/gradle-api/8.0/com/android/build/api/dsl/BuildFeatures#buildConfig()

File: feature_search/build.gradle.kts [hunk 13]
```diff
@@ -27,15 +28,14 @@ android {
   buildFeatures {
     // Enables Jetpack Compose for this module
     compose = true
+    buildConfig = true
```
[the remark this answers is not in the document]
Comment (ricardo-frango) [reply feature_search/build.gradle.kts:31]: Apparently on the the new AGP 8.0 the default value has been changed to false. I had to change it to true.

https://developer.android.com/reference/tools/gradle-api/8.0/com/android/build/api/dsl/BuildFeatures#buildConfig()

File: feature_editorial/build.gradle.kts (hunk 7 above)
Comment (ricardo-frango) [reply feature_editorial/build.gradle.kts:33]: Apparently on the the new AGP 8.0 the default value has been changed to false. I had to change it to true.

https://developer.android.com/reference/tools/gradle-api/8.0/com/android/build/api/dsl/BuildFeatures#buildConfig()

File: install-manager/build.gradle.kts (hunk 9 above)
Comment (ricardo-frango) [reply install-manager/build.gradle.kts:8]: After updating the AGP, I had the error that namespace was not specified. so I had to move it from the manifest to build.gradle file.
From AGP 7.3, we should stop specifying the package in the manifest.

https://developer.android.com/guide/topics/manifest/manifest-element#package

File: install-manager/src/main/AndroidManifest.xml
Comment (ricardo-frango) [reply install-manager/src/main/AndroidManifest.xml:1]: After updating the AGP, I had the error that namespace was not specified. so I had to move it from the manifest to build.gradle file.
From AGP 7.3, we should stop specifying the package in the manifest.

https://developer.android.com/guide/topics/manifest/manifest-element#package

File: download-view/src/main/java/cm/aptoide/pt/download_view/presentation/DownloadView.kt
[the remark this answers is not in the document]
Comment (ricardo-frango) [reply download-view/src/main/java/cm/aptoide/pt/download_view/presentation/DownloadView.kt:1]: reverted the formatting changes but I had to change the packaged names because if you check the code, the "ic_icon_error" is actually from the "aptoide_ui" module and not from the "download_view" module. The AGP might have some changes on building and this icon is only found in the "aptoide_ui" module.
That's why I had to change to package of the icon.

File: feature_apps/src/main/java/cm/aptoide/pt/feature_apps/presentation/AppGraphicView.kt
Comment (ricardo-frango) [reply feature_apps/src/main/java/cm/aptoide/pt/feature_apps/presentation/AppGraphicView.kt:1]: Reverted the formatting changes.
I had to change the package name for icons "ic_icon_star" because it does not exist in the package "cm.aptoide.pt.feature_apps" probably due to changes in the AGP build

File: feature_apps/src/main/java/cm/aptoide/pt/feature_apps/presentation/AppGridView.kt
Comment (ricardo-frango) [reply feature_apps/src/main/java/cm/aptoide/pt/feature_apps/presentation/AppGridView.kt:1]: I had to change the package from "cm.aptoide.pt.feature_apps.R" to "cm.aptoide.pt.aptoide_ui.R" because the icons "ic_appcoins_logo" and "ic_icon_star" do not exist in the "feature_apps" module. This might be related to changes in the the AGP build.
The icon "ic_placeholder" was already being used from the package "cm.aptoide.pt.aptoide_ui.R" so I just removed that unnecessary code.

File: feature_appview/src/main/java/cm/aptoide/pt/feature_appview/presentation/AppViewScreen.kt
[the remark this answers is not in the document]
Comment (ricardo-frango) [reply feature_appview/src/main/java/cm/aptoide/pt/feature_appview/presentation/AppViewScreen.kt:1]: I had to change the package names because the icons in this file do not exist in the "feature_appview" module. So I needed to change the package name.

File: feature_appview/src/main/java/cm/aptoide/pt/feature_appview/presentation/OtherVersionsView.kt
[the remark this answers is not in the document]
Comment (ricardo-frango) [reply feature_appview/src/main/java/cm/aptoide/pt/feature_appview/presentation/OtherVersionsView.kt:1]: fixed

File: feature_appview/src/main/java/cm/aptoide/pt/feature_appview/presentation/RelatedContentView.kt
Comment (ricardo-frango) [reply feature_appview/src/main/java/cm/aptoide/pt/feature_appview/presentation/RelatedContentView.kt:1]: reverted the formatting changes.
The package was not changed because the icons "ic_placeholder" and "ic_views" only exist in the "aptoide_ui" module. Due to changes in the AGP build, I had to update the package

File: feature_appview/src/main/java/cm/aptoide/pt/feature_appview/presentation/ReviewsView.kt
[the remark this answers is not in the document]
Comment (ricardo-frango) [reply feature_appview/src/main/java/cm/aptoide/pt/feature_appview/presentation/ReviewsView.kt:1]: reverted the formatting changes.
Kept the change in the package name because the package "cm.aptoide.pt.feature_appview.R" does not have the icons in this file. This might be related with changes in the new AGP build system.

File: feature_editorial/src/main/java/cm/aptoide/pt/feature_editorial/presentation/EditorialScreen.kt
[the remark this answers is not in the document]
Comment (ricardo-frango) [reply feature_editorial/src/main/java/cm/aptoide/pt/feature_editorial/presentation/EditorialScreen.kt:1]: I had to change the package name because this package does not contains these images anymore. This might be related with changes in the AGP build system.

File: feature_editorial/src/main/java/cm/aptoide/pt/feature_editorial/presentation/EditorialsCardView.kt
Comment (ricardo-frango) [reply feature_editorial/src/main/java/cm/aptoide/pt/feature_editorial/presentation/EditorialsCardView.kt:1]: reverted the formatting changes.
I kept the changes of the package name because does icons do not exist in the package "cm.aptoide.pt.feature_editorial.R" does not contain the drawables in this file.
This might be related with changes in the AGP build system.

File: feature_report_app/src/main/java/cm/aptoide/pt/feature_report_app/presentation/ReportAppView.kt
Comment (ricardo-frango) [reply feature_report_app/src/main/java/cm/aptoide/pt/feature_report_app/presentation/ReportAppView.kt:1]: reverted the formatting changes.
I had to keep the changes to the package names because the package "cm.aptoide.pt.feature_report_app.R" does not contain the drawables used in this file.
This might be related with changes in the AGP build system.

**Review by G0retZ — APPROVED** [review]
